hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveMetastore.java

```diff
@@ -282,6 +282,8 @@ private void initConf(HiveConf conf, int port, boolean directSql) {
     // Setting this to avoid thrift exception during running Iceberg tests outside Iceberg.
     conf.set(
         HiveConf.ConfVars.HIVE_IN_TEST.varname, HiveConf.ConfVars.HIVE_IN_TEST.getDefaultValue());
+
+    conf.set("datanucleus.connectionPoolingType", "HikariCP");
```

Comment:
Spark 4.0 removed dependency on com.jolbox:bonecp.  If not set the cp type, it will default to `com.jolbox:bonecp` and we will get
```
Exception in thread "TThreadPoolServer WorkerProcess-5" java.lang.NoClassDefFoundError: com/jolbox/bonecp/BoneCPConfig
	at org.apache.hadoop.hive.metastore.txn.TxnHandler.setupJdbcConnectionPool(TxnHandler.java:3156)
	at org.apache.hadoop.hive.metastore.txn.TxnHandler.setConf(TxnHandler.java:260)
```